sympy/simplify/simplify.py

```diff
@@ -2540,8 +2541,8 @@ def recurse(arg, **kwargs):
 
     expr = sympify(expr)
 
-    if not isinstance(expr, Basic) or expr.is_Atom or expr in (
-            exp_polar(0), exp_polar(1)):
+    if (not isinstance(expr, Basic) or isinstance(expr, MatrixSymbol) or (
+            expr.is_Atom or expr in (exp_polar(0), exp_polar(1)))):
```

Comment:
Ok. Looking through, `MatrixSymbol` isn't recursed into in `simplify` currently anyway, so if this stops it from crashing, I'm ok with it for now.